core/domain/collection_services_test.py

```diff
@@ -648,6 +647,24 @@ def _create_search_query(self, terms, categories):
                 '"%s"' % category for category in categories]) + ')'
         return query
 
+    def test_get_collection_summaries_matching_ids(self):
+        summaries = collection_services.get_collection_summaries_matching_ids([
+            self.COL_ID_0, self.COL_ID_1, self.COL_ID_2, 'nonexistent'])
+        self.assertEqual(summaries[0].title, 'Bridges in England')
+        self.assertEqual(summaries[1].title, 'Introduce Oppia')
+        self.assertEqual(summaries[2].title, 'Introduce Interactions in Oppia')
+        self.assertIsNone(summaries[3])
```

Comment:
This is odd -- if I am reading the code correctly the idea is to omit None entries, right?

Reply:
`get_collection_summaries_subscribed_to` omits `None` entries, `get_collection_summaries_matching_ids` keeps `None` entries. (`test_get_collection_summaries_matching_ids` was already in the codebase before my PR but I have added new test for that method).